### ISSUE TITLE
make dist fails on latest master (error while building cycle_finder jar)

### ISSUE THREAD

**Issue by myabc** [google/j2objc#2325]
Encountering the following error while building `cycle_finder`:
```
building cycle_finder jar
../translator/src/main/java/com/google/devtools/j2objc/types/GeneratedExecutableElement.java:23: error: package com.google.errorprone.annotations does not exist
import com.google.errorprone.annotations.CanIgnoreReturnValue;
                                        ^
../translator/src/main/java/com/google/devtools/j2objc/types/GeneratedExecutableElement.java:217: error: cannot find symbol
  @CanIgnoreReturnValue
   ^
  symbol:   class CanIgnoreReturnValue
  location: class GeneratedExecutableElement
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
2 errors
make[1]: *** [build_result/cycle_finder.jar] Error 1
make: *** [cycle_finder_dist] Error 2
```

To reproduce:

```
make dist
```

My environment:

```
 make print_environment
Locale: en_GB.UTF-8
Darwin Mondiale-Sechs 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:11:08 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T8122 arm64
Xcode 15.3 Build version 15E204a
Apple clang version 15.0.0 (clang-1500.3.9.4)
Target: arm64-apple-darwin23.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

Environment:
ANDROID_AVD_HOME=/Users/alexbcoles/.android/avd
ANDROID_HOME=/Users/alexbcoles/Library/Android/sdk
ANDROID_NDK_HOME=/opt/homebrew/share/android-ndk
COLORFGBG=7;0
COLORTERM=truecolor
COMMAND_MODE=unix2003
GEM_HOME=/Users/alexbcoles/.gem/ruby/3.2.2
GEM_PATH=/Users/alexbcoles/.gem/ruby/3.2.2:/Users/alexbcoles/.rubies/ruby-3.2.2/lib/ruby/gems/3.2.0
GEM_ROOT=/Users/alexbcoles/.rubies/ruby-3.2.2/lib/ruby/gems/3.2.0
HOME=/Users/alexbcoles
HOMEBREW_CELLAR=/opt/homebrew/Cellar
HOMEBREW_PREFIX=/opt/homebrew
HOMEBREW_REPOSITORY=/opt/homebrew
INFOPATH=/opt/homebrew/share/info:
ITERM_PROFILE=Default
ITERM_SESSION_ID=w0t0p0:4BD124AD-6914-488E-A756-CD733A3100B0
J2OBJC_ARCHS=macosx iphone64 iphone64e watchosv7k watchos64 watchsimulator watchsimulator64 simulator simulator64 maccatalyst appletvos appletvsimulator macosx64 maccatalyst64
JAVA_HOME=/opt/homebrew/Cellar/openjdk@11/11.0.23/libexec/openjdk.jdk/Contents/Home
LANG=en_GB.UTF-8
LC_ALL=en_US.UTF-8
LC_TERMINAL=iTerm2
LC_TERMINAL_VERSION=3.4.23
LESS=-R
LOGNAME=alexbcoles
LSCOLORS=Gxfxcxdxbxegedabagacad
LS_COLORS=di=1;36:ln=35:so=32:pi=33:ex=31:bd=34;46:cd=34;43:su=30;41:sg=30;46:tw=30;42:ow=30;43
MAKEFLAGS=
MAKELEVEL=1
MANPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/share/man:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/usr/share/man:/Applications/Xcode.app/Contents/Developer/usr/share/man:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/share/man:/opt/homebrew/opt/nvm/versions/node/v20.12.2/share/man:/opt/homebrew/share/man::
MFLAGS=
NVM_BIN=/opt/homebrew/opt/nvm/versions/node/v20.12.2/bin
NVM_CD_FLAGS=-q
NVM_DIR=/opt/homebrew/opt/nvm
NVM_INC=/opt/homebrew/opt/nvm/versions/node/v20.12.2/include/node
PAGER=less
PATH=/Users/alexbcoles/.gem/ruby/3.2.2/bin:/Users/alexbcoles/.rubies/ruby-3.2.2/lib/ruby/gems/3.2.0/bin:/Users/alexbcoles/.rubies/ruby-3.2.2/bin:/Users/alexbcoles/.pyenv/shims:/opt/homebrew/opt/nvm/versions/node/v20.12.2/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/Users/alexbcoles/.cargo/bin:/Users/alexbcoles/Library/Android/sdk/tools:/Users/alexbcoles/Library/Android/sdk/cmdline-tools/latest/bin:/Users/alexbcoles/Library/Android/sdk/platform-tools:/opt/homebrew/Cellar/openjdk@11/11.0.23/libexec/openjdk.jdk/Contents/Home/bin
PROTOBUF_ROOT_DIR=/opt/homebrew
PWD=/Users/alexbcoles/git-repos/j2objc
PYENV_ROOT=/Users/alexbcoles/.pyenv
PYENV_SHELL=zsh
RUBYOPT=
RUBY_ENGINE=ruby
RUBY_ROOT=/Users/alexbcoles/.rubies/ruby-3.2.2
RUBY_VERSION=3.2.2
SHELL=/bin/zsh
SHLVL=2
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.KOHxcGxVPt/Listeners
TERM=xterm-256color
TERM_PROGRAM=iTerm.app
TERM_PROGRAM_VERSION=3.4.23
TERM_SESSION_ID=w0t0p0:4BD124AD-6914-488E-A756-CD733A3100B0
TMPDIR=/var/folders/5c/fwpc8q792bv0g_7n2t7t_yfh0000gn/T/
USER=alexbcoles
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
ZSH=/Users/alexbcoles/.oh-my-zsh
```

**Comment by myabc** [comment]
Removing two lines from the following source file in `translator` works around the issue:
```diff
diff --git a/translator/src/main/java/com/google/devtools/j2objc/types/GeneratedExecutableElement.java b/translator/src/main/java/com/google/devtools/j2objc/types/GeneratedExecutableElement.java
index 3a3bbb23b..994fb7fb3 100644
--- a/translator/src/main/java/com/google/devtools/j2objc/types/GeneratedExecutableElement.java
+++ b/translator/src/main/java/com/google/devtools/j2objc/types/GeneratedExecutableElement.java
@@ -20,7 +20,6 @@ import com.google.devtools.j2objc.ast.DebugASTPrinter;
 import com.google.devtools.j2objc.util.ElementUtil;
 import com.google.devtools.j2objc.util.NameTable;
 import com.google.devtools.j2objc.util.TypeUtil;
-import com.google.errorprone.annotations.CanIgnoreReturnValue;
 import java.lang.annotation.Annotation;
 import java.util.Collection;
 import java.util.List;
@@ -214,7 +213,6 @@ public class GeneratedExecutableElement extends GeneratedElement implements Exec
     return originalElement;
   }
 
-  @CanIgnoreReturnValue
   public GeneratedExecutableElement setOriginalElement(ExecutableElement element) {
     originalElement = element;
     return this;
```

**Comment by tomball** [comment]
The issue is the same as [how j2objc/translator's build was fixed](https://github.com/google/j2objc/commit/1d5ade80c648acb857cefc9f87b14d820f536b33): add `$(ERROR_PRONE_ANNOTATIONS_JAR)` to the `INTERNAL_DEPS` list in j2objc/cycle_finder/Makefile. No need to edit any Java sources.

I'm working on a commit to fix this problem, as well as failures when building with Java 22 (no need to do so since Java 11 works fine, but some developers get antsy being on older versions).

**Comment by tomball** [comment]
PR #2326 out for review, should be submitted Monday.

**Comment by tomball** [comment]
Fixed in PR #2326.
